src/X.Serilog.Sinks.Telegram/Batch/Rules/BatchSizeRule.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading;
+
+namespace X.Serilog.Sinks.Telegram.Batch.Rules;
+
+internal class BatchSizeRule : IRule
+{
+    private readonly ILogsQueueAccessor _accessContext;
+    private readonly int _batchSize;
+
+    public BatchSizeRule(ILogsQueueAccessor accessContext, int batchSize)
+    {
+        _batchSize = batchSize;
+        _accessContext = accessContext;
+    }
+
+    public Task<bool> IsPassedAsync(CancellationToken cancellationToken)
```

Comment:
Candidate for ValueTask as well, as this code seems to not do any async stuff inside